tensorflow/python/ops/rnn.py

```diff
@@ -810,7 +810,8 @@ def _time_step(time, output_ta_t, state):
 
   # Unpack final output if not using output tuples.
   if in_graph_mode:
-    final_outputs = tuple(ta.stack() for ta in output_final_ta)
+    final_outputs = tuple(
+        ta.gather(range(time_steps)) for ta in output_final_ta)
```

Comment:
this assumes time_steps is statically known; which in general it is not.

Reply:
Can you explain why it is not?  The time steps parameter (graph compile time) is both the size of the output tensor array and the loop counter.  It comes from the shape of the input sequence, which is known.

I'm not suggesting that you take this change, if there is an alternative XLA compatible dynamic_rnn in the pipeline.  Just curious why it isn't static.